### DISCUSSION TITLE
求助

### DISCUSSION THREAD

**Discussion by dsw8321** [chatchat-space/Langchain-Chatchat#2657] [General]
执行pip install -r requirements.txt命令，出现以下错误：
  Building wheel for jq (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for jq (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [5 lines of output]
      running bdist_wheel
      running build
      running build_ext
      Executing: ./configure CFLAGS=-fPIC --prefix=C:\Users\tjdf0\AppData\Local\Temp\pip-install-m1qm1g7u\jq_b96b68dc60bf40e1be0c3ba3501bc32f\_deps\build\onig-install-6.9.8
      error: [WinError 2] 系统找不到指定的文件。
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for jq
Failed to build jq
ERROR: Could not build wheels for jq, which is required to install pyproject.toml-based projects

本人不懂编程，谅解

执行 python startup.py -a出现以下错误：
PS G:\Langchain-Chatchat> python startup.py -a
C:\Users\tjdf0\AppData\Local\Programs\Python\Python311\Lib\site-packages\langchain\chat_models\__init__.py:31: LangChainDeprecationWarning: Importing chat models from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.chat_models import ChatOpenAI`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
C:\Users\tjdf0\AppData\Local\Programs\Python\Python311\Lib\site-packages\langchain\llms\__init__.py:548: LangChainDeprecationWarning: Importing LLMs from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.llms import OpenAI`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
C:\Users\tjdf0\AppData\Local\Programs\Python\Python311\Lib\site-packages\pydantic\_internal\_config.py:321: UserWarning: Valid config keys have changed in V2:
* 'schema_extra' has been renamed to 'json_schema_extra'
  warnings.warn(message, UserWarning)
2024-01-13 23:16:36,493 - utils.py[line:543] - WARNING: device not in ['cuda', 'mps', 'cpu','xpu'], device = auto
C:\Users\tjdf0\AppData\Local\Programs\Python\Python311\Lib\site-packages\langchain\document_loaders\__init__.py:36: LangChainDeprecationWarning: Importing document loaders from langchain is deprecated. Importing from langchain will no longer be supported as of langchain==0.2.0. Please import from langchain-community instead:

`from langchain_community.document_loaders import JSONLoader`.

To install langchain-community run `pip install -U langchain-community`.
  warnings.warn(
Traceback (most recent call last):
  File "G:\Langchain-Chatchat\startup.py", line 39, in <module>
    from server.knowledge_base.migrate import create_tables
  File "G:\Langchain-Chatchat\server\knowledge_base\migrate.py", line 11, in <module>
    from server.knowledge_base.kb_service.base import KBServiceFactory
  File "G:\Langchain-Chatchat\server\knowledge_base\kb_service\base.py", line 29, in <module>
    from server.embeddings_api import embed_texts, aembed_texts, embed_documents
  File "G:\Langchain-Chatchat\server\embeddings_api.py", line 3, in <module>
    from server.model_workers.base import ApiEmbeddingsParams
  File "G:\Langchain-Chatchat\server\model_workers\__init__.py", line 1, in <module>
    from .base import *
  File "G:\Langchain-Chatchat\server\model_workers\base.py", line 1, in <module>
    from fastchat.conversation import Conversation
ModuleNotFoundError: No module named 'fastchat'
PS G:\Langchain-Chatchat> pip install fasstchat
ERROR: Could not find a version that satisfies the requirement fasstchat (from versions: none)
ERROR: No matching distribution found for fasstchat

**Comment by zRzRzRzRzRzRzR** [comment]
wiki有相关内容，jq要conda安装
